### DISCUSSION TITLE
Incorporating Edge Features into GATConv for edge-level prediction

### DISCUSSION THREAD

**Discussion by koongjl** [pyg-team/pytorch_geometric#9055] [Q&A] [answered]
I am encountering this problem, where I need to predict the label of the 4000 input edges, but the model only output 1000 predictions. 
![image](https://github.com/pyg-team/pytorch_geometric/assets/154297147/0bc5e848-6dec-4656-ad89-a2b962c69f4c)

this is my train data:
![image](https://github.com/pyg-team/pytorch_geometric/assets/154297147/731ddbf9-6133-48fd-a398-7a5060833b6c)

this is the code:
![image](https://github.com/pyg-team/pytorch_geometric/assets/154297147/7b8ebcf1-9c5d-4400-b1ae-c0d3a4dd5946)
![image](https://github.com/pyg-team/pytorch_geometric/assets/154297147/07eaa1a0-f21b-47ad-af57-c1be4b63a988)

**Comment by EdisonLeeeee** [comment, marked as answer]
You need an additional module (decoder) to decode the edge representations from the node representations. For example:
```python
class EdgeDecoder(torch.nn.Module):
    def __init__(self,):
        super().__init__()
        self.lin1 = torch.nn.Linear(32, 64)
        self.lin2 = torch.nn.Linear(64, 1)

    def forward(self, x, edge_index):
        x1 = x[edge_index[0]]
        x2 = x[edge_index[1]]

        h = torch.cat([x1, x2], dim=1)
        h = self.lin1(h).relu()
        return self.lin2(h).sigmoid()

# ...
decoder = EdgeDecoder()
# ...
pedictions = model(train_data)
predictions = decoder(predictions, train_data.edge_index)
# ...

```
